### ISSUE TITLE
this is a very clever idea !

### ISSUE THREAD

**Issue by aktxyz** [GobieGenerator/Gobie#20]

**Comment by Siphonophora** [comment]
@aktxyz Thanks, glad you like it. 

Anything in particular you would use it for? Still very much looking for feedback on how to apply this concept.

**Comment by aktxyz** [comment]
I have not tried it out yet.  It does look like it would be perfect for the various source generators I am using ... but these are working fine now so don't want to spend the time redoing them.  Next time.

**Comment by Siphonophora** [comment]
Yeah, no worries about trying it now. At the moment, it crashes too much to be worth having anyone test the demo.

**Comment by jeffward01** [comment]
> @aktxyz Thanks, glad you like it.
> 
> Anything in particular you would use it for? Still very much looking for feedback on how to apply this concept.

Personally I use this for source generation. I have spent hours and hours working with incremental source generators and its quite a painful experience.

Often in code, we have patterns like in DDD or the repository pattern, where you must create the same class over and over again, like CRUD domain events or something similar.  

I use Source Generators for this!! 

I have followed your work for a while!!  Congrats about the release!
